pkg/importer/vddk-datasource.go

```diff
@@ -244,10 +338,62 @@ func (vs *VDDKDataSource) Transfer(path string) (ProcessingPhase, error) {
 
 // TransferFile is called to transfer the data from the source to the file passed in.
 func (vs *VDDKDataSource) TransferFile(fileName string) (ProcessingPhase, error) {
-	err := qemuOperations.ConvertToRawStream(vs.GetURL(), destinationFile)
+	size, err := vs.NbdHandle.GetSize()
 	if err != nil {
-		klog.Infof("Failed to convert disk: %s\n", err)
+		klog.Errorf("Unable to get size from libnbd handle: %v", err)
 		return ProcessingPhaseError, err
 	}
+
+	sink, err := newVddkDataSink(destinationFile, size)
+	if err != nil {
+		return ProcessingPhaseError, err
+	}
+	defer sink.Close()
```

Comment:
This is just curiousity on my part, does the sink have like a flush or something at the end to ensure that the data has been written to the actual storage device, and not stuck somewhere in a cache somewhere? I ask because we had a bug where we had RWX pvc, and tried to immediately start the VM on a different node, and not everything was written to the device, and starting the VM failed due to a corrupted disk.

Reply:
That's a good idea, I added a final file.Sync in the normal implementation (not the test mock one) of sink.Close.